src/app/settings/admin/bucket-settings/destinations/component.ts

```diff
@@ -0,0 +1,176 @@
+  }
+
+  displayWarning(): boolean {
+    return !!this._hasOldData() && !this.hasDestinations();
+  }
+
+  hasCredentials(destination: BackupDestination): boolean {
+    return !!destination && !!destination.credentials;
+  }
+
+  addDestination(): void {
+    const dialogConfig: MatDialogConfig = {
+      data: {
+        title: 'Add Destination',
+        mode: Mode.Add,
+        confirmLabel: 'Add',
+        seed: this.seed,
+      },
+    };
+
+    this._matDialog.open(DestinationDialog, dialogConfig);
+  }
+
+  editDestination(destination: BackupDestination): void {
+    const dialogConfig: MatDialogConfig = {
+      data: {
+        title: 'Edit Destination',
+        mode: Mode.Edit,
+        confirmLabel: 'Edit',
+        seed: this.seed,
+        destination: destination,
+      },
+    };
+
+    this._matDialog.open(DestinationDialog, dialogConfig);
+  }
+
+  editCredentials(destination: BackupDestination): void {
+    const dialogConfig: MatDialogConfig = {
+      data: {
+        seed: this.seed,
+        destination: destination,
+      },
+    };
+
+    this._matDialog.open(EditCredentialsDialog, dialogConfig);
+  }
+
+  deleteDestination(destination: BackupDestination): void {
+    const dialogConfig: MatDialogConfig = {
+      disableClose: false,
+      hasBackdrop: true,
+      data: {
+        title: 'Delete Destination',
+        message: `Delete <b>${destination.destinationName}</b> destination permanently?`,
+        warning: 'Associated backups and snapshots will not be usable after deleting this destination.',
+        confirmLabel: 'Delete',
+      } as ConfirmationDialogConfig,
+    };
+
+    const destinations: Destinations = this.seed.spec.etcdBackupRestore.destinations;
+    delete destinations[destination.destinationName];
+
+    const configuration: AdminSeed = this.seed;
+    configuration.spec.etcdBackupRestore.destinations = destinations;
+
+    this._matDialog
+      .open(ConfirmationDialogComponent, dialogConfig)
+      .afterClosed()
+      .pipe(filter(isConfirmed => isConfirmed))
+      .pipe(switchMap(_ => this._datacenterService.patchAdminSeed(configuration.name, configuration)))
+      .pipe(take(1))
+      .subscribe(_ => {
+        this._notificationService.success(`The destination ${destination.destinationName} was deleted`);
+        this._datacenterService.refreshAdminSeeds();
+      });
+  }
+
+  private _mapToArray(): void {
+    if (this.hasDestinations()) {
+      for (const [k, v] of Object.entries(this.seed.spec.etcdBackupRestore.destinations)) {
+        this.destinations.push({
+          destinationName: k,
+          bucketName: v.bucketName,
+          endpoint: v.endpoint,
+          credentials: v.credentials,
+        });
+      }
+    }
+  }
+
+  private _hasOldData(): boolean {
+    return (
+      !!this.seed &&
+      !!this.seed.spec &&
+      !!this.seed.spec.backupRestore &&
+      (!!this.seed.spec.backupRestore.s3BucketName || !!this.seed.spec.backupRestore.s3Endpoint)
```

Comment:
As above.

Reply:
This can still be improved. First line is not needed if you add "?" to the second.